ui/src/components/feedback/Toast.js

```diff
@@ -39,11 +40,34 @@ export class Toast extends LitElement {
         ToastUtils.clearToast(this);
     }
 
+    /**
+     * Emits an event that results in an attempt to manually
+     * resubmit the last failed audio upload.
+     */
+    handleReuploadAudio() {
```

Comment:
This could be made more generic by instead passing in an optional "action" `mwc-icon` name.

So when dispatching the toast, you can send a `retry` icon name with it, and when that button is clicked an `action-${iconName}` event could be dispatched.

Not sure if you have time for the refactor. Up to you.

Reply:
adjusted